Page colour

will-moore opened the pull request:
Allows you to change the background colour of the pages.
See https://trello.com/c/55EzQq2e/14-set-page-colour

<img width="1186" alt="screen shot 2017-04-29 at 23 57 49" src="https://cloud.githubusercontent.com/assets/900055/25559696/cc5f656c-2d37-11e7-87f3-f133fd26d92f.png">


To test:
 - Create a new figure with a few labels on the outside of the panel, at least 1 white, 1 black and 1 other color. White labels should be rendered Black (so you can see them against White page).
 - Go to File -> Page setup and pick page colour.
 - Page should update with chosen colour when Page_Setup dialog is "OK" & closed.
 - Try setting the page colour to Black.
 - Black labels should be rendered White (but not if moved to be within the panel instead of on the outside of the panel).
 - Saving and refreshing/reloading figure should preserve page colour.
 - Export of figure to PDF or TIFF should include coloured page.

NB: I'll need to upload script before the export is tested

snoopycrimecop commented:
Conflicting PR. Removed from build [FIGURE-merge#426](https://ci.openmicroscopy.org/job/FIGURE-merge/426/). See the [console output](https://ci.openmicroscopy.org/job/FIGURE-merge/426/consoleText) for more details.
Possible conflicts:
  - PR #209 will-moore 'Markdown labels'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - PR #210 will-moore 'Export to omero'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

pwalczysko commented:
Bug: Export as Tiff does not preserve the feature where black labels on black bckgr are white. See screenshot, on the left is the figure, on the right the exported tiff. 
![screen shot 2017-06-14 at 10 21 56](https://user-images.githubusercontent.com/2478303/27125208-77cad6a8-50eb-11e7-81c0-20f28dd89774.png)

pwalczysko commented:
All the rest works as expected, except https://github.com/ome/omero-figure/pull/211#issuecomment-308375269. Discussed the bug with @bramalingam - if this is intended behaviour, then maybe rather remove the feature completely (feature = being clever about black on black and white on white).

will-moore commented:
To test last commit:

 - create both Black and White labels on the outside of panels
 - Choose black page - all labels should appear white and be exported as white
 - Choose white page - all labels should appear black and be exported as black
 - Choose coloured page - labels should appear and export black or white as created

pwalczysko commented:
Works as expected. Ready to merge.

jburel commented on src/js/models/panel_model.js:
```diff
@@ -233,11 +233,11 @@
-                        labels[j].color == "FFFFFF") {
-                    labels[j].color = "000000";
-                }
+                // // check that we're not adding a white label outside panel (on a white background)
```
remove if it is no longer needed

jburel commented on omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:
```diff
@@ -130,7 +130,8 @@ def __init__(self, canvas, panel, page, crop, page_height):
 
-    def get_rgb(self, color):
+    @staticmethod
+    def get_rgb(color):
```
moving to @staticmethod means that this could be used everywhere and no need to have the same method line 424

will-moore commented:
Addressed comments from @jburel.
To test:
 - export a PDF and check that shapes (ellipse, rectangle, line & arrow) on a panel are the correct colour.

pwalczysko commented:
Works as expected. Ready to merge.
